geo/src/algorithm/line_measures/metric_spaces/euclidean.rs

```diff
@@ -0,0 +1,77 @@
+use super::super::Distance;
+use crate::{GeoFloat, Point};
+
+/// Operations on the [Euclidean plane] measure distance with the pythagorean formula -
+/// what you'd measure with a ruler.
+///
+/// If you have lon/lat points, use the [`Haversine`], [`Geodesic`], or other [metric spaces] -
+/// Euclidean methods will give nonsense results.
+///
+/// Alternatively, you *can* use lon/lat points with Euclidean methods if you first [`Transform`]
```

Comment:
I understand the point you're trying to make, but if you transform lon/lat coordinates they're not lon/lat coordinates anymore. Could we say something like "If you wish to use Euclidean operations with `lon`/`lat` coordinates these must first be be transformed using the `transform` / `transform_crs_to_crs` methods or their immutable variants. Use of these requires the `proj` feature"?

Reply:
I'm OK with your phrasing — I've updated it.

For my own understanding, are you trying to avoid the usage of the term "projection"?

Reply:
Not consciously, but yes: If you're working with non-Euclidean coordinates, you're _probably_ not that interested in projection. If you were, you wouldn't be using measures and algorithms that are slower and more complex than their 2D counterparts.

This is all part of a larger endeavour, in the sense that `geo` is trying to be a practical tool with which to do real work. A great deal of that work can be done by transforming geographic coordinates into geodetic coordinates, giving you access to the simpler, more robust abstractions in the Euclidean plane.

But we also know (because the abstractions and algorithms now exist in `geo`  because people have asked for / contributed them) that some work is more suited to spherical geometry.

As it stands, I feel like our docs and the way that some of our algorithms are organised don't demarcate that relationship very well; In _my_ mind, we're "Euclidean and Cartesian by default, but if you need spherical geometry and algorithms we've got some of those".

Reply:
>  In my mind, we're "Euclidean and Cartesian by default, but if you need spherical geometry and algorithms we've got some of those".

Yeah, I think that accurately reflects georust/geo. I think it also pretty accurately reflects most historical GIS software.

> This is all part of a larger endeavour, in the sense that geo is trying to be a practical tool with which to do real work. 

💯 ❤️ 

> If you were [interested in projection], you wouldn't be using measures and algorithms that are slower and more complex than their 2D counterparts.

With "The Web" as a publishing platform (e.g. webmaps, geojson) and more global datasets like OSM, using lon/lat coordinates in algorithms is increasingly not a conscious decision of the analyst/cartographer/person as "the best", but rather just happens to be the format that their source data is in, or that their presentation layer demands.

Having talked through this a bit, I think that's the group I'm trying to connect with, and often am implicitly biased towards in my documentation and API design.